fastapi_jwks/models/types.py

```diff
@@ -25,3 +25,66 @@ class JWKSConfig(BaseModel):
 class JWTTokenInjectorConfig(BaseModel):
     payload_field: str = "payload"
     token_field: str = "raw_token"
+
+
+class JWTHeader(BaseModel):
+    model_config: ClassVar[ConfigDict] = ConfigDict(extra="allow")
+
+    alg: Annotated[str, Field(description="Algorithm used for signing")]
+    typ: Annotated[
+        str | None,
+        Field(description="Type of token", examples=["JWT"]),
+    ] = None
+    cty: Annotated[str | None, Field(description="Content type")] = None
+    kid: Annotated[str | None, Field(description="Key ID")] = None
+    x5u: Annotated[str | None, Field(description="X.509 URL")] = None
+    x5c: Annotated[list[str] | None, Field(description="X.509 Certificate Chain")] = (
+        None
+    )
+    x5t: Annotated[
+        str | None, Field(description="X.509 Certificate SHA-1 Thumbprint")
+    ] = None
+    x5tS256: Annotated[
+        str | None, Field(description="X.509 Certificate SHA-256 Thumbprint")
+    ] = None
+
+
+class JWK(BaseModel):
+    """JSON Web Key"""
+
+    model_config: ClassVar[ConfigDict] = ConfigDict(extra="allow", frozen=True)
+
+    kty: Annotated[str, Field(description="Key Type")]
+    use: Annotated[str | None, Field(description="Public key use")] = None
+    key_ops: Annotated[list[str] | None, Field(description="Key operations")] = None
```

Comment:
Does the RFC specify if camel or pascal case is used?

Reply:
I suppose only snake case is valid in this case